### ISSUE TITLE
Language crash in .json files

### ISSUE THREAD

**Issue by BaileyJM02** [bartosz-antosik/vscode-spellright#127]
When using `en_GB` on most files, such as `.md`, `.php` and `.html` the English (GB) language works fine. When I use it on `.json` it crashes and only works on English (US).

**Comment by BaileyJM02** [comment]
Maybe something to do with the fact it has 1000+ lines?

**Comment by bartosz-antosik** [comment]
Could you tell me please what platform are you taking about?

**Comment by bartosz-antosik** [comment]
What do you mean by "crashes"?

And also what is the value of `spellright.groupDictionaries` in your config?

**Comment by BaileyJM02** [comment]
## Information
- Platform: Linux 
  - Ubuntu 17.x
- Vs Code:
  - Version 1.21.1
  - Commit 79b44aa704ce542d8ca4a3cc44cfca566e7720f1
  - Date 2018-03-14T14:46:30.761Z
  - Shell 1.7.9
  - Renderer 58.0.3029.110
  - Node 7.9.0
  - Architecture x64
- `spellright.groupDictionaries = true` (default)

## Video
> http://vps.bailey.guru/spellright-bug.mp4

**Comment by BaileyJM02** [comment]
**[Correction]** It works with `.json` files, just not mine (They're over 1000 lines)

**Comment by BaileyJM02** [comment]
Okay, I have a fix.

You have to use JSON with comments, `jsonc` otherwise, if you use `json` syntax and a file that *contains* comments without declaring it. Your plugin crashes.

**Comment by BaileyJM02** [comment]
![image](https://user-images.githubusercontent.com/19824075/37558133-3c95f81a-2a07-11e8-8b17-7aa978c43bf5.png)

**Comment by bartosz-antosik** [comment]
Thanks for the update! But that's interesting in many aspects:

1) What do you consider comment in JSON? I think formally there should not be comments in JSON files. 
2) `jsonc` is not internally supported by Spell Right. So it should ask you about which parser to use for this type of file: a) Has it? b) What parser have you selected?
3) I would be very grateful if you could maybe please synthesize a short JSON example showing the problem? Because I really do not think the size is a problem...

**Comment by bartosz-antosik** [comment]
Oh, I have missed the video. It shows how does the crash look like. I have never seen anything alike. Thanks for the video!

**Comment by BaileyJM02** [comment]
I've currently not at my computer anymore... I've left the house but when I get back I'll send everything you request :D

**Comment by bartosz-antosik** [comment]
Thank you!

**Comment by BaileyJM02** [comment]
1. I see a comment as such:
   ```JSON
   {
      "data":"something", //comment
   }
   ```
2. Perhaps you should add it ;) 
   1. Yes it did.
   2. This is what is within "User Settings", I picked plain...
      ```JSON
      "spellright.parserByClass": {
         "jsonc": {
         "parser": "plain"
       }
      ```
3. The snippet: (We'll use my settings)
   - Working
      ```JSON
      "spellright.language": "en_GB",
      "spellright.documentTypes": [
          "markdown",
          "latex",
          "plaintext",
          "jsonc"
      ],
      "spellright.parserByClass": {
          "jsonc": {
          "parser": "plain"
          }
      },
      ```
   - Not working
      ```JSON
      "spellright.language": "en_GB", //Say put a comment here (btw this works with en_US)
      "spellright.documentTypes": [
          "markdown",
          "latex",
          "plaintext", //I told you what this is
          "jsonc"
      ],
      "spellright.parserByClass": {
          "jsonc": {
          "parser": "plain" //perhaps it's because it's within an array :thinking:
      },
      ```

**Comment by bartosz-antosik** [comment]
I have tried to reproduce this problem but I have failed:

![image](https://user-images.githubusercontent.com/20492548/37602831-811d67f2-2b8d-11e8-8c2a-b7f521ef57bc.png)

Above is of course Windows but Spell Right works in HUNSPELL mode so it should be more or less very similar to what happens on Linux.

But I think that using plaintext spell checker for JSON with comments is wrong so as first step I will try to include `jsonc` among types supported by proper parser and it you could please after upgrade try to remove the `"jsonc": { "parser": "plain" }` from `"spellright.parserByClass"` setting and then tell me whether this work would be a great help.

**Comment by BaileyJM02** [comment]
Yeah, @mention me when you update and I'm on Linux so that might be the problem.

**Comment by BaileyJM02** [comment]
And can you try with JSON please, (not JSONc) as using JSONc was the fix.

**Comment by bartosz-antosik** [comment]
It's already been updated. Version 2.1.32.

**Comment by BaileyJM02** [comment]
It crashed on Version `2.1.32` also... I'm uploading a gist of the actual file I have been using... along with the translation file.

**Comment by BaileyJM02** [comment]
Please view [the gist](https://gist.github.com/BaileyJM02/5ab7170a00b9852f4063daf2290f4c42).

**Comment by bartosz-antosik** [comment]
Would be very grateful if you could either confirm or deny my debug observation supported hypothesis that the problem is emojis in JSON. 

Could you delete emojis from the JSON and see if it crashes?

**Comment by BaileyJM02** [comment]
Actually, that makes sense as I tried to add an emoji earlier and it also crashed.

Sorry about all the confusion, I'm guessing it's emojis.

**Comment by bartosz-antosik** [comment]
No problem. Glad we caught it. Looks like 100% emojis. I will soon release a patched version. Thanks for cooperation on this.

**Comment by bartosz-antosik** [comment]
This should be corrected in version 2.1.34. Please have a look and tell me whether it does.
